src/main/java/teammates/common/util/RecaptchaVerifier.java

```diff
@@ -0,0 +1,66 @@
+package teammates.common.util;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import org.apache.http.client.utils.URIBuilder;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+/**
+ * Used to handle the verification of the user's reCAPTCHA response.
+ *
+ * @see <a href="https://developers.google.com/recaptcha/docs/verify">reCAPTCHA user response verification API</a>
+ */
+public class RecaptchaVerifier {
+
+    /** The Google reCAPTCHA API URL to verify the response token. */
+    public static final String VERIFY_URL = "https://www.google.com/recaptcha/api/siteverify";
+
+    /** The shared secret key between the TEAMMATES site and reCAPTCHA. */
+    public static final String SECRET_KEY = Config.CAPTCHA_SECRET_KEY;
+
+    private static final Logger log = Logger.getLogger();
+
+    /**
+     * Returns true if the {@code captchaResponse} token is verified successfully.
+     * @param captchaResponse The user's captcha response from the client side
+     * @return true if the {@code captchaResponse} is verified successfully, and false if an exception occurs or if the
+     *         API request fails
+     */
+    public boolean isVerificationSuccessful(String captchaResponse) {
+        if (captchaResponse == null || captchaResponse.isEmpty()) {
+            return false;
+        }
+
+        try {
+            String response = getApiResponse(captchaResponse);
+
+            log.info("reCAPTCHA API response: " + response);
+            JsonObject responseInJson = JsonUtils.parse(response).getAsJsonObject();
+
+            if (responseInJson.has("error-codes")) {
+                JsonArray errorCodes = responseInJson.get("error-codes").getAsJsonArray();
+                log.warning("Error codes during reCAPTCHA verification: " + errorCodes.toString());
+            }
+
+            return Boolean.parseBoolean(responseInJson.get("success").toString());
+        } catch (IOException e) {
+            log.severe("reCAPTCHA request failure: " + e.getMessage());
+            return false;
+        } catch (Exception e) {
+            log.severe("reCAPTCHA request failure: " + e.getStackTrace());
+            return false;
+        }
```

Comment:
for the two catch blocks above, you would want to take a look at the log msgs see what they give you. right now i m getting some weird stuff:
![image](https://user-images.githubusercontent.com/28526038/53821627-25efb480-3fa9-11e9-8846-be19f81be392.png)

Reply:
Updated

Reply:
![image](https://user-images.githubusercontent.com/28526038/53851871-02555a00-3ffb-11e9-8dd5-be84762b4a57.png)

how does this look like? if you want the trace you can do ```e.printStackTrace()``` after you log.

Reply:
That causes a PMD violation...

Reply:
`e.printStackTrace()` causes a PMD violation.

Updated to use java `Logger#log(..., Throwable)` instead

Reply:
looks good!